processing/src/main/java/org/apache/druid/java/util/http/client/response/BytesFullResponseHolder.java

```diff
@@ -42,13 +43,16 @@ public BytesFullResponseHolder addChunk(byte[] chunk)
   }
 
   @Override
-  public byte[] getContent()
+  public byte[] getContent() throws BufferOverflowException
   {
-    int size = 0;
+    long size = 0L;
     for (byte[] chunk : chunks) {
       size += chunk.length;
+      if (size > Integer.MAX_VALUE) {
+        throw new BufferOverflowException();
```

Comment:
can you please add the long size in the message? Also, if you can catch this exception in RemoteTaskActionClient, you can rethrow an exception with more user friendly message.